### PR TITLE
:recycle: Refactor Daemon Protocol to use JSON for Communication

### PR DESCRIPTION
**Pull request by dkelosky**
Depends on https://github.com/zowe/imperative/pull/681, see description there

Signed-off-by: Dan Kelosky <dkelosky@gmail.com>

### CODE REVIEW

**Comment by zFernand0** [comment]
This PR will likely fix https://github.com/zowe/zowe-cli/issues/1195

File: zowex/src/main.rs [hunk 1]
```diff
@@ -181,18 +200,20 @@ fn run_daemon_command(mut args: String) -> std::io::Result<()> {
         let daemon_proc_info = is_daemon_running();
 
         // when not running, start it.
+        // if daemon_proc_info.is_running == false {
```
Comment (awharn) [inline zowex/src/main.rs:203]: Can this line be removed?

**Review by awharn — COMMENTED** [review]
I only have one comment so far. 😄

File: zowex/src/main.rs (hunk 1 above)
Comment (gejohnston) [reply zowex/src/main.rs:203]: I second Andrew's opinion that the commented line should be removed. It looks like something that was temporarily commented out during testing, and then left by accident afterward.

**Comment by gejohnston** [comment]
I merged the zowex automated testing updates into this branch and ran the zowex nectos/act unit tests and integration tests.

Both the original header-style communication and the JSON-style communication gave the same test results:

```
Unit test:
Test Suites: 260 passed, 260 total
Tests:       1229 passed, 1229 total
Snapshots:   445 passed, 445 total
```

```
Integration tests:
Test Suites: 99 failed, 1 skipped, 99 of 100 total
Tests:       235 failed, 2 skipped, 237 total
Snapshots:   105 failed, 105 total
```

These results imply that the JSON communication introduces no new misbehavior into the zowe daemon. Thus, I believe that this PR can be safely merged into our @next branch as soon as its corresponding zowe/imperative#681 has been merged.

**Comment by zFernand0** [comment]
I saw the same number of failing tests on `next`, but after attempting to use an older version of imperative and doing an `npm ci`, I got the expected result again.

All these numbers are for integration test results

Yesterday results: (same as Gene's)
(every test fails with all the headers being printed to `stdout` at the end of every command)
```
Test Suites: 99 failed, 1 skipped, 99 of 100 total
Tests:       235 failed, 2 skipped, 237 total
Snapshots:   105 failed, 105 total
```

`next` results: (after doing  `npm ci` and attempting to use an older version of imperative)
(only `stderr > stdout` issues)
```
Test Suites: 64 failed, 1 skipped, 35 passed, 99 of 100 total
Tests:       132 failed, 2 skipped, 103 passed, 237 total
Snapshots:   37 failed, 96 passed, 133 total
```

`next-zowe-daemon` results: (JSON communication)
(lest errors) 🥳 
```
Test Suites: 20 failed, 1 skipped, 79 passed, 99 of 100 total
Tests:       38 failed, 2 skipped, 197 passed, 237 total
Snapshots:   26 failed, 127 passed, 153 total
```

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**